### PR TITLE
Add class.jetpack-rest-sync

### PR DESCRIPTION
**Pull request by enejb**
**Do NOT merge**

Improves to posts syncing. 

#### Changes proposed in this Pull Request:
The idea here is to improve post syncing when data related to the post changes. 
So that we can better reconstruct the rest api endpoint.

### CODE REVIEW

File: tests/php/test_class.jetpack-sync.php [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+		Jetpack_Rest_Sync::init();
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+	}
+
+	public function test_sync_new_post() {
+		$new_post = array(
```
Comment (lezama) [inline tests/php/test_class.jetpack-sync.php]: I think we could extract this to a helper as we are repeating it a lot of times.

**Comment by enejb** [comment]
https://github.com/Automattic/jetpack/commit/4129a26b09694d0ddbf5dafd466f7d5273c1737d 
Should fix the failing tests.

File: jetpack.php [hunk 2]
```diff
@@ -26,6 +26,8 @@
 defined( 'JETPACK__API_BASE' )               or define( 'JETPACK__API_BASE', 'https://jetpack.wordpress.com/jetpack.' );
 defined( 'JETPACK_PROTECT__API_HOST' )       or define( 'JETPACK_PROTECT__API_HOST', 'https://api.bruteprotect.com/' );
 defined( 'JETPACK__WPCOM_JSON_API_HOST' )    or define( 'JETPACK__WPCOM_JSON_API_HOST', 'public-api.wordpress.com' );
+defined( 'JETPACK_WPCOM_JSON_API__BASE' )    or define( 'JETPACK_WPCOM_JSON_API__BASE', JETPACK__WPCOM_JSON_API_HOST . '/rest/v1' );
```
Comment (gravityrail) [inline jetpack.php]: why doesn't this constant have a double-underscore like the one above it?

Comment (enejb) [inline jetpack.php]: Good catch :+1:

**Comment by jeherve** [comment]
@enejb Should we close this now that you've created #3527?

**Comment by lezama** [comment]
> @enejb Should we close this now that you've created #3527?

:+1:

**Pull request closed without merging**